src/app/ReadClient.h

```diff
@@ -84,11 +100,17 @@ class ReadClient : public Messaging::ExchangeDelegate
 
     enum class ClientState
     {
-        Uninitialized = 0,     ///< The client has not been initialized
-        Initialized,           ///< The client has been initialized and is ready for a SendReadRequest
-        AwaitingInitialReport, ///< The client is waiting for initial report
+        Uninitialized = 0,         ///< The client has not been initialized
+        Initialized,               ///< The client has been initialized and is ready for a SendReadRequest
+        AwaitingInitialReport,     ///< The client is waiting for initial report
+        AwaitingSubscribeResponse, ///< The client is waiting for subscribe response
+        SubscriptionIdle,          ///< The client is maintaining subscription
     };
 
+    bool IsClientForSubscription(uint64_t aSubscriptionId)
```

Comment:
Would recommend a less wordier version:

```suggestion
    bool IsMatchingClient(uint64_t aSubscriptionId)
```

Reply:
done, thanks